lib/mpl_toolkits/mplot3d/axes3d.py

```diff
@@ -2641,6 +2641,103 @@ def calc_arrow(uvw, angle=15):
 
     quiver3D = quiver
 
+    def stem(self, *args, **kwargs):
+        """
+        Create a 3D stem plot.
+        
+        Call signatures::
+          stem3(x, y, z, linefmt='b-', markerfmt='bo', basefmt='r-', 
+                zdir={'x'|'y'|'z'|'-x'|'-y'})
+        
+        By default, stem plots vertical lines (using *linefmt*) in the z direction
+        at each *x* and *y* location from the baseline to *z*, and places a marker there
+        using *markerfmt*. By default, the baseline at the xy-plane is plotted using
+        *basefmt*.
+        
+        *x*, *y, and *z* values are required and must be arrays of the same length.
+        
+        If no zdir value is provided, then it is set to default and no rotation occurs.
+        
+        Return value is a tuple (*markerline*, *stemlines*, *baseline*).
+        
+        .. seealso::
+            This `document <http://www.mathworks.com/help/techdoc/ref/stem3.html>`_
+            for details.
+        
+        **Example:**
+        .. plot:: /mplot3d/stem3d_demo.py
+        """
+
+        from matplotlib.container import StemContainer
+
+        had_data = self.has_data()
+
+        remember_hold = self._hold
+        if not self._hold:
+            self.cla()
+        self.hold(True)
+
+        # Extract the required values
+        x, y, z = [np.asarray(i) for i in args[:3]]
+        args = args[3:]
+
+        # Popping some defaults
+        try:
+            linefmt = kwargs.pop('linefmt', args[0])
+        except IndexError:
+            linefmt = kwargs.pop('linefmt', 'b-')
+        try:
+            markerfmt = kwargs.pop('markerfmt', args[1])
+        except IndexError:
+            markerfmt = kwargs.pop('markerfmt', 'bo')
+        try:
+            basefmt = kwargs.pop('basefmt', args[2])
+        except IndexError:
+            basefmt = kwargs.pop('basefmt', 'r-')
+        try:
+            zdir = kwargs.pop('zdir', args[3])
+        except IndexError:
+            zdir = kwargs.pop('zdir', 'z')
+
+        bottom = kwargs.pop('bottom', None)
+        label = kwargs.pop('label', None)
+                    
+        if bottom is None:
+            bottom = 0
+
+        stemlines = []
+        jx, jy, jz = art3d.juggle_axes(x, y, z, zdir)
+        
+        #plot the baseline in the appropriate plane
+        for i in range(len(x)-1):
+            
+            baseline, = Axes.plot(self, [x[i], x[i+1]], [y[i], y[i+1]],
```

Comment:
Probabyl better to use a `LineCollection` here.  Or at least directly create `artist.Line2D`.

You should also set the color here to eliminate the need for the `setp` calls in your examples and tests.